### ISSUE TITLE
Nix support in Apx does not function as documented. 

### ISSUE THREAD

**Issue by big-edd** [Vanilla-OS/apx#241]
Hello, 

As suggested here..  ..  
https://vanillaos.org/2023/02/07/apx-now-supports-nix-packages.html
I first ran this..  ..  
apx init --nix 
/nix/ and ~/.nix/ were created as expected. 

But when I ran this..  ..  
apx install --nix htop 
The result was this..  ..  
  ERROR   exec: "nix": executable file not found in $PATH

Cheers, 
Eddie.

**Comment by mirkobrombin** [comment]
Nix is no more a feature in Apx v2, adding the v1 label

**Comment by kbdharun** [comment]
Hello,

If Nix didn't initialize correctly, try running:

```bash
sudo systemctl disable nix.mount ensure-nix-dir.service ensure-nix-own.service
rm -rf ~/.nix*
sudo rm -rf /nix
```

Then try initializing again, this should fix it.

> Nix is no more a feature in Apx v2, adding the v1 label

Regarding the context for the above comment check out this [Devlog](https://vanillaos.org/2023/06/11/vanilla-os-orchid-devlog.html).

TLDR. We are moving Nix implementation in apx (direct integration of Nix shell, not a container) to VSO (Vanilla System Operator) [not distro-agnostic] as we making apx's future version a dev tool focused on containers, so If you are in other distros than Vanilla I would suggest installing Nix shell directly.

I am closing this now, if you still face the issue feel free to reopen it.
